### ISSUE TITLE
C++ equivalent api

### ISSUE THREAD

**Issue by adderly** [Cutehacks/duperagent#5]
It will be cool to have a version for c++, with an smaller memory footprint 💃

**Comment by jrbarron** [comment]
@adderly I agree this would be cool and you're not the first person to ask for it :) If I had to start from scratch today I would maybe have made an effort to do this the first time around. Doing it at this point would be a fairly big task though so I don't see it happening anytime soon. Maybe for version 2.x. If someone else wants to consider doing this though, I'll happily discuss the API and implementation here. I'll leave this task open for discussion and voting purposes.

**Comment by adderly** [comment]
@jrbarron Well let's get the discussion started, i might help with the things i need. And if anyone gets on the bandwagon things can be more straight forward.
